### ISSUE TITLE
stage2: Compiler panic in `Value.writeToMemory` when computing comptime SHA-256 hash

### ISSUE THREAD

**Issue by topolarity** [ziglang/zig#13335]
### Zig Version

0.10.0-dev.4622+f28e4e03e

### Steps to Reproduce and Observed Behavior

```
const std = @import("std");
const Hasher = std.crypto.hash.sha2.Sha256;
const testfile = @embedFile("testfile");
test {
    comptime
    {
        @setEvalBranchQuota(100_000_000);
        var digest: [Hasher.digest_length]u8 = undefined;
        _ = Hasher.hash(testfile, &digest, .{});
        //std.log.err("digest: {}", .{std.fmt.fmtSliceHexLower(digest[0..])});
        @compileLog(digest);
    }
}
```

"testfile" is a 1KB file of random bytes generated using `dd if=/dev/urandom of=testfile bs=1K count=1`

Output:
```console
Semantic Analysis [115] writeIntBig... thread 6675544 panic: reached unreachable code
/Users/toadman667/repos/zig/lib/std/debug.zig:278:14: 0x102f62107 in std.debug.assert (zig2)
    if (!ok) unreachable; // assertion failure
             ^
/Users/toadman667/repos/zig/lib/std/math/big/int.zig:2225:15: 0x1032fa1c9 in std.math.big.int.Const.writeTwosComplement (zig2)
        assert(buffer.len >= byte_count);
              ^
/Users/toadman667/repos/zig/src/value.zig:1226:43: 0x103a85f60 in value.Value.writeToMemory (zig2)
                bigint.writeTwosComplement(buffer, bits, abi_size, target.cpu.arch.endian());
                                          ^
/Users/toadman667/repos/zig/src/Sema.zig:25529:38: 0x103732cb9 in Sema.storePtrVal (zig2)
            operand_val.writeToMemory(operand_ty, sema.mod, buffer[reinterpret.byte_offset..]);
                                     ^
/Users/toadman667/repos/zig/src/Sema.zig:25437:33: 0x10372a510 in Sema.storePtr2 (zig2)
            try sema.storePtrVal(block, src, ptr_val, operand_val, elem_ty);
                                ^
/Users/toadman667/repos/zig/src/Sema.zig:4705:26: 0x103710eaf in Sema.zirStoreNode (zig2)
    return sema.storePtr2(block, src, ptr, ptr_src, operand, operand_src, air_tag);
                         ^
/Users/toadman667/repos/zig/src/Sema.zig:1099:38: 0x1034a25e1 in Sema.analyzeBodyInner (zig2)
                try sema.zirStoreNode(block, inst);
                                     ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1321:62: 0x1034a82f4 in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:658:30: 0x10348610b in Sema.analyzeBody (zig2)
    _ = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                             ^
/Users/toadman667/repos/zig/src/Sema.zig:6454:33: 0x1039ffe9b in Sema.analyzeCall (zig2)
                sema.analyzeBody(&child_block, fn_info.body) catch |err| switch (err) {
                                ^
/Users/toadman667/repos/zig/src/Sema.zig:5985:32: 0x10367315b in Sema.zirCall (zig2)
        return sema.analyzeCall(block, func, func_src, call_src, modifier, ensure_result_used, resolved_args, bound_arg_src);
                               ^
/Users/toadman667/repos/zig/src/Sema.zig:774:62: 0x1034972b0 in Sema.analyzeBodyInner (zig2)
            .call                         => try sema.zirCall(block, inst),
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1321:62: 0x1034a82f4 in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:658:30: 0x10348610b in Sema.analyzeBody (zig2)
    _ = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                             ^
/Users/toadman667/repos/zig/src/Sema.zig:6454:33: 0x1039ffe9b in Sema.analyzeCall (zig2)
                sema.analyzeBody(&child_block, fn_info.body) catch |err| switch (err) {
                                ^
/Users/toadman667/repos/zig/src/Sema.zig:5985:32: 0x10367315b in Sema.zirCall (zig2)
        return sema.analyzeCall(block, func, func_src, call_src, modifier, ensure_result_used, resolved_args, bound_arg_src);
                               ^
/Users/toadman667/repos/zig/src/Sema.zig:774:62: 0x1034972b0 in Sema.analyzeBodyInner (zig2)
            .call                         => try sema.zirCall(block, inst),
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1321:62: 0x1034a82f4 in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1427:62: 0x1034a50bd in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1321:62: 0x1034a82f4 in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1298:62: 0x1034a8083 in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1321:62: 0x1034a82f4 in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:658:30: 0x10348610b in Sema.analyzeBody (zig2)
    _ = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                             ^
/Users/toadman667/repos/zig/src/Sema.zig:6454:33: 0x1039ffe9b in Sema.analyzeCall (zig2)
                sema.analyzeBody(&child_block, fn_info.body) catch |err| switch (err) {
                                ^
/Users/toadman667/repos/zig/src/Sema.zig:5985:32: 0x10367315b in Sema.zirCall (zig2)
        return sema.analyzeCall(block, func, func_src, call_src, modifier, ensure_result_used, resolved_args, bound_arg_src);
                               ^
/Users/toadman667/repos/zig/src/Sema.zig:774:62: 0x1034972b0 in Sema.analyzeBodyInner (zig2)
            .call                         => try sema.zirCall(block, inst),
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:675:45: 0x10348e01d in Sema.analyzeBodyBreak (zig2)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/Users/toadman667/repos/zig/src/Sema.zig:1321:62: 0x1034a82f4 in Sema.analyzeBodyInner (zig2)
                const break_data = (try sema.analyzeBodyBreak(block, inline_body)) orelse
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:658:30: 0x10348610b in Sema.analyzeBody (zig2)
    _ = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                             ^
/Users/toadman667/repos/zig/src/Sema.zig:6454:33: 0x1039ffe9b in Sema.analyzeCall (zig2)
                sema.analyzeBody(&child_block, fn_info.body) catch |err| switch (err) {
                                ^
/Users/toadman667/repos/zig/src/Sema.zig:5952:42: 0x103672a34 in Sema.zirCall (zig2)
            break :b try sema.analyzeCall(block, func, func_src, call_src, modifier, ensure_result_used, resolved_args, bound_arg_src);
                                         ^
/Users/toadman667/repos/zig/src/Sema.zig:774:62: 0x1034972b0 in Sema.analyzeBodyInner (zig2)
            .call                         => try sema.zirCall(block, inst),
                                                             ^
/Users/toadman667/repos/zig/src/Sema.zig:5086:34: 0x103a24db8 in Sema.resolveBlockBody (zig2)
        if (sema.analyzeBodyInner(child_block, body)) |_| {
                                 ^
/Users/toadman667/repos/zig/src/Sema.zig:5069:33: 0x10371d86a in Sema.zirBlock (zig2)
    return sema.resolveBlockBody(parent_block, src, &child_block, body, inst, &label.merges);
                                ^
/Users/toadman667/repos/zig/src/Sema.zig:1307:69: 0x1034a43cc in Sema.analyzeBodyInner (zig2)
                if (!block.is_comptime) break :blk try sema.zirBlock(block, inst);
                                                                    ^
/Users/toadman667/repos/zig/src/Sema.zig:658:30: 0x10348610b in Sema.analyzeBody (zig2)
    _ = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                             ^
/Users/toadman667/repos/zig/src/Module.zig:5652:21: 0x1032a1de4 in Module.analyzeFnBody (zig2)
    sema.analyzeBody(&inner_block, fn_info.body) catch |err| switch (err) {
                    ^
/Users/toadman667/repos/zig/src/Module.zig:4344:40: 0x103281bce in Module.ensureFuncBodyAnalyzed (zig2)
            var air = mod.analyzeFnBody(func, sema_arena) catch |err| switch (err) {
                                       ^
/Users/toadman667/repos/zig/src/Compilation.zig:3063:42: 0x10311f2d3 in Compilation.processOneJob (zig2)
            module.ensureFuncBodyAnalyzed(func) catch |err| switch (err) {
                                         ^
/Users/toadman667/repos/zig/src/Compilation.zig:3001:30: 0x10310d3f6 in Compilation.performAllTheWork (zig2)
            try processOneJob(comp, work_item);
                             ^
/Users/toadman667/repos/zig/src/Compilation.zig:2329:31: 0x1031058fe in Compilation.update (zig2)
    try comp.performAllTheWork(main_progress_node);
                              ^
/Users/toadman667/repos/zig/src/main.zig:3323:20: 0x103095880 in main.updateModule (zig2)
    try comp.update();
                   ^
/Users/toadman667/repos/zig/src/main.zig:3008:17: 0x102fd97de in main.buildOutputType (zig2)
    updateModule(gpa, comp, hook) catch |err| switch (err) {
                ^
/Users/toadman667/repos/zig/src/main.zig:236:31: 0x102f712d5 in main.mainArgs (zig2)
        return buildOutputType(gpa, arena, args, .zig_test);
                              ^
/Users/toadman667/repos/zig/src/stage1.zig:56:24: 0x102f70a06 in main (zig2)
        stage2.mainArgs(gpa, arena, args) catch unreachable;
                       ^
???:?:?: 0x1174bd52d in ??? (???)
???:?:?: 0x0 in ??? (???)
[1]    2481 abort      ./build/zig2 test testct.zig --zig-lib-dir lib
```

### Expected Behavior

Should compute the same hash as at runtime 
(`74b7d714d7c401efbb6a4c276101f8e1e93e74962310744c180bc9352fded2d7` for my testfile)

**Comment by Vexu** [comment]
Duplicate/related #12937

**Comment by mlugg** [comment]
Was resolved by 1e6d7f7
